derive_builder/examples/validator_fn.rs

```diff
@@ -0,0 +1,37 @@
+//! This example illustrates the use of `validator` to add a pre-build validation
+//! step.
+
+#[macro_use]
+extern crate derive_builder;
+
+#[derive(Builder, Debug, PartialEq)]
+#[builder(build_fn(validator="LoremBuilder::validate"))]
```

Comment:
Consider a family of builders which have the common requirement that their server IP address must be organization-internal. It would be easy to implement a short trait with one method to expose the server IP and a `validate` function with a default implementation.